qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,396 @@
+        """
+        self.__max_iter = value
+
+    @property
+    def alpha(self) -> float:
+        """Returns the alpha parameter value (percentage of population selected to update
+        probabilistic model)"""
+        return self.__alpha
+
+    @alpha.setter
+    def alpha(self, value: float):
+        """
+        Sets the alpha parameter (percentage of individuals selected to update the probabilistic
+        model)
+
+        Args:
+            value: Percentage [0,1] of generation selected to update the probabilistic model.
+        """
+        self.__alpha = value
+
+    @property
+    def n_variables(self) -> int:
+        """Returns the number of variables desired to be optimized"""
+        return self.__n_variables
+
+    @n_variables.setter
+    def n_variables(self, value: int):
+        """
+        Sets the number of variables to be optimized by the UMDA.
+
+        Args:
+            value: Number of variables to be optimized.
+        """
+        self.__n_variables = value
+
+    @property
+    def dead_iter(self) -> int:
+        """Returns the stopping criteria: the number of iterations with no improvement, after
+        which the algorithm converges"""
+        return self.__dead_iter
+
+    @dead_iter.setter
+    def dead_iter(self, value: int):
+        """
+        Sets the stopping criteria: the number of iterations with no improvement after which the
+        algorithm converges and stops.
+
+        Args:
+            value: Number of iterations with no improvement after which the algorithm converges.
+        """
+        self.__dead_iter = value
+
+    @property
+    def best_cost_global(self) -> float:
+        """Returns the best individual cost found until the moment"""
+        return self.__best_cost_global
+
+    @property
+    def best_ind_global(self):
+        """Returns the best individual instance found until the moment"""
+        return self.__best_ind_global
+
+    @property
+    def generation(self):
+        """Returns the actual generation of solutions"""
+        return self.__generation
+
+    @generation.setter
+    def generation(self, value: np.array):
+        """
+        Sets the generation to be evaluated. The user may want to initialize the algorithm from a
+        specific set of solutions.
+
+        Args:
+            value: set of solutions to be set as generation.
+        """
+        self.__generation = value
+
+    @property
+    def history(self) -> list:
+        """Returns the best cost found in each iteration during runtime"""
+        return self.__history
+
+    @property
+    def settings(self) -> Dict[str, Any]:
+        return {
+            "max_iter": self.max_iter,
+            "alpha": self.alpha,
+            "dead_iter": self.dead_iter,
+            "size_gen": self.size_gen,
+            "n_variables": self.n_variables,
+            "best_cost_global": self.best_cost_global,
+            "best_ind_global": self.best_ind_global,
```

Comment:
My understanding is that settings reflects the constructor parameters so that these values can be queried/serialized to generate a new remote instance for the runtime. Hence it should be limited to just these @Cryoris - please correct me if needed.

Reply:
That is True. This two parameters have been removed.